paimon-core/src/main/java/org/apache/paimon/rest/RESTCatalog.java

```diff
@@ -434,6 +528,23 @@ protected GetTableResponse getTableResponse(Identifier identifier)
         }
     }
 
+    protected SuccessResponse dropPartitionMetadata(
+            Identifier identifier, Map<String, String> partitions)
+            throws TableNoPermissionException {
+        try {
+            DropPartitionRequest request = new DropPartitionRequest(partitions);
+            return client.delete(
+                    resourcePaths.partitions(
+                            identifier.getDatabaseName(), identifier.getTableName()),
+                    request,
+                    headers());
+        } catch (NoSuchResourceException ignore) {
+            return new SuccessResponse();
```

Comment:
I think here should throw table not exist exception?

Reply:
when the partition's metadata doesn't exist in metadata, there may still be data in the filesystem. So we ignore this exception when dropping the partition.

Reply:
should return false instead of NoSuchResourceException?